ql/src/java/org/apache/hadoop/hive/ql/exec/ReplCopyTask.java

```diff
@@ -177,6 +184,24 @@ protected int execute(DriverContext driverContext) {
         listBW.close();
       }
 
+      // If the srcMap is not empty which means we made the list of files for distCp. If there are files from different
+      // filesystems, then the map have multiple entries.
+      if (!srcMap.isEmpty()) {
+        boolean success = true;
+        for (final HashMap.Entry<FileSystem, List<Path>> entry : srcMap.entrySet()) {
+          FileSystem actualSrcFs = entry.getKey();
+          List<Path> srcPaths = entry.getValue();
+          if (!doCopy(toPath, dstFs, srcPaths, actualSrcFs, conf)) {
+            console.printError("Failed to copy: " + srcPaths.size()
+                    + " files to: '" + toPath.toString() + "'");
+            success = false;
```

Comment:
if we encounter an error, can we return the error flag earlier rather than trying to run the loop for other file systems, before returning. 

also rather than doing double negation with ! can we rename variables and reorder logic such that its straight forward reading the code.